others/postfix_evaluation.cpp

```diff
@@ -0,0 +1,157 @@
+/**
+ * @file
+ * @brief Evaluation of Postfix Expression
+ * @author [Darshana Sarma](https://github.com/Darshana-Sarma)
+ * @details
+ * Create a stack to store operands (or values).
+ * Scan the given expression and do following for every scanned element.
+ * If the element is a number, push it into the stack
+ * If the element is a operator, pop operands for the operator from stack.
+ * Evaluate the operator and push the result back to the stack
+ * When the expression is ended, the number in the stack is the final answer
+ */
+#include <algorithm>  // for all_of
+#include <array>      // for std::array
+#include <cassert>    // for assert
+#include <iostream>   // for io operations
+// Maximum size for stack
+const int MAX_SIZE = 20;
+
+// Defining stack
+std::array<float, MAX_SIZE> stack;
+
+// Index of top value in stack array
+int stackTop;
+
+/**
+ * Pushing operand, also called the number in the array to the stack
+ * @param operand float value from the input array or evaluation
+ * @returns none
+ */
+void push(float operand) {
+    stackTop++;
+    stack[stackTop] = operand;
+}
+
+/**
+ * Popping operand, also called the number from the stack
+ * @param none
+ * @returns operand float on top of stack
+ */
+float pop() {
+    float operand = stack[stackTop];
+    stackTop--;
+    return operand;
+}
+
+/**
+ * Checks if scanned string is a number
+ * @param s scanned string
+ * @returns bool boolean value if string is number
+ */
+bool is_number(const std::string &s) {
+    return !s.empty() && std::all_of(s.begin(), s.end(), ::isdigit);
+}
+
+/**
+ * Evaluate answer using given operands and operation
+ * @param a float
+ * @param b float
+ * @param operation operation to be performed with respective floats
+ * @returns none
+ */
+void evaluate(float a, float b, const std::string &operation) {
+    float c = 0;
+    const char *op = operation.c_str();
+    switch (*op) {
+        case '+':
+            c = a + b;
+            push(c);
+            break;
+
+        case '-':
+            c = a - b;
+            push(c);
+            break;
+
+        case '*':
+            c = a * b;
+            push(c);
+            break;
+
+        case '/':
+            c = a / b;
+            push(c);
+            break;
+
+        default:
+            std::cout << "Operator not defined\n";
+            break;
+    }
+}
+
+/**
+ * Postfix Evaluation algorithm to compute the value from given input array
+ * @param input Array of characters consisting of numbers and operations
+ * @param N Array size
+ * @returns stack[stackTop] returns the top value from the stack
+ */
+template <std::size_t N>
+float postfix_evaluation(std::array<std::string, N> input) {
+    stackTop = -1;
+    int j = 0;
+
+    while (j < N) {
+        std::string scan = input[j];
+        if (is_number(scan)) {
+            push(std::stof(scan));
```

Comment:
add the string header

`#include <string>`

From what I can tell, the `std::stof` is the trouble maker. If adding the string header doesn't resolve it then there might be an issue with the compile flag.

Reply:
for [reference](http://www.cplusplus.com/reference/string/stod/)